Index: include/llvm/ExecutionEngine/ExecutionEngine.h
===================================================================
--- include/llvm/ExecutionEngine/ExecutionEngine.h	(revision 116193)
+++ include/llvm/ExecutionEngine/ExecutionEngine.h	(working copy)
@@ -183,6 +183,22 @@
   virtual void addModule(Module *M) {
     Modules.push_back(M);
   }
+
+  //===--------------------------------------------------------------------===//
+  // Module iterator forwarding functions
+  //
+  typedef SmallVector<Module*, 1> ModuleVecType;
+   typedef ModuleVecType::const_iterator const_module_iterator;
+  const_module_iterator modules_begin() const { return Modules.begin(); }
+  const_module_iterator modules_end  () const { return Modules.end();   }
+
+  size_t                modules_size() const { return Modules.size();  }
+  bool                  modules_empty() const { return Modules.empty(); }
+  const Module         *modules_front() const { return Modules.front(); }
+  const Module         *modules_back() const { return Modules.back();  }
+  const Module         *modules_at(unsigned i) const { return Modules[i];  }
+
+   
   
   //===----------------------------------------------------------------------===//
 
