Index: tools/Makefile
===================================================================
--- tools/Makefile	(revision 129351)
+++ tools/Makefile	(working copy)
@@ -10,7 +10,7 @@
 LEVEL := ..
 
 # Build clang if present.
-OPTIONAL_PARALLEL_DIRS := clang
+OPTIONAL_PARALLEL_DIRS := clang cling
 
 # Build LLDB if present. Note LLDB must be built last as it depends on the
 # wider LLVM infrastructure (including Clang).
